### PR TITLE
Add support for x-only keys in descriptor

### PR DESCRIPTION
**Pull request by sanket1729**
.

### CODE REVIEW

File: src/miniscript/context.rs [hunk 1]
```diff
@@ -45,6 +45,9 @@ pub enum ScriptContextError {
     /// Only Compressed keys allowed under current descriptor
     /// Segwitv0 fragments do not allow uncompressed pubkeys
     CompressedOnly(String),
+    /// XOnly keys are only allowed in Tap context
+    /// The first element is key, and second element is current script context
+    XOnlyKeysNotAllowed(String, String),
```
Comment (apoelstra) [inline src/miniscript/context.rs:50]: In cbd02d4910e56c01995152c27624a5f9d3da06e5:

I think the second `String` could be a `&'static str`

File: src/descriptor/key.rs [hunk 2]
```diff
@@ -427,7 +460,10 @@ impl DescriptorPublicKey {
         self
     }
 
-    /// Computes the public key corresponding to this descriptor key
+    /// Computes the public key corresponding to this descriptor key.
+    /// When deriving from an XOnlyPublicKey, it adds the default 0x02 y-ordinate
```
Comment (apoelstra) [inline src/descriptor/key.rs:464]: In ce1425e3b8bfef7aad4c0ca0003ecf6381463b47:

typo `y-ordinate`

File: src/descriptor/mod.rs [hunk 3]
```diff
@@ -557,10 +557,48 @@ impl Descriptor<DescriptorPublicKey> {
...
+    /// In most cases, you would want to use [`Self::derived_descriptor`] directly to obtain
+    /// a [`Descriptor<bitcoin::PublicKey>`]
     pub fn derive(&self, index: u32) -> Descriptor<DescriptorPublicKey> {
         self.translate_pk2_infallible(|pk| pk.clone().derive(index))
     }
 
+    /// Derive a [`Descriptor`] with a concrete [`bitcoin::PublicKey`] at a given index
+    /// Removes all extended pubkeys and wildcards from the descriptor and only leaves
+    /// concrete [`bitcoin::PublicKey`]. All [`crate::XOnlyKey`]s are converted to [`bitcoin::PublicKey`]
+    /// by adding a default(0x02) y-ordinate. For [`crate::descriptor::Tr`] descriptor,
```
Comment (apoelstra) [inline src/descriptor/mod.rs:570]: In f2d7bdeed7309eada7c2c8a536e0a0f2776866bd:

`y-ordinate` again. Is this a term that I don't know? :)

**Comment by apoelstra** [comment]
Done reviewing f2d7bdeed7309eada7c2c8a536e0a0f2776866bd

**Comment by sanket1729** [comment]
Addressed the comments.

**Review by apoelstra — APPROVED** [review]
ACK 602e94f7a53fa7834897d7bb977486411e973a21

**Pull request merged**